Web MIDI API support

atsushieno opened the pull request:
This changeset brings in MIDI I/O support which is based on Web MIDI API (via a patched RtMidi which is also going to be PR-ed to the upstream).

So far rtmidi is a single-source module it is pretty easy to add to the codebase (like what ROLI does for juce_box2d).

DemoRunner is also updated to back up additional exports for this change.

Sleepwalking commented:
Hello Atsushi,

Sorry for this very long wait to get this PR merged. We finally had sometime to update `juce_emscripten` after the launch event. What a roller coaster ride!

I added a "JUCE_WEBMIDI" macro check for turning off this feature (it's enabled by default), because some people may not like the permission popup.

atsushieno commented:
I have rebased all the changes off of more recent JUCE master (months ago) for easier upstream merging in the future, and added some Projucer support in this branch. Feel free to take anything there. https://github.com/atsushieno/JUCE/commits/juce_emscripten

(There is juce_emscripten_juce6 branch too but it was an unfinished attempt.)
